src/main/java/teammates/storage/sqlapi/FeedbackSessionsDb.java

```diff
@@ -113,6 +132,28 @@ public void deleteFeedbackSession(FeedbackSession feedbackSession) {
         }
     }
 
+    /**
+     * Soft-deletes a specific feedback session by its name and course id.
+     *
+     * @return Soft-deletion time of the feedback session.
+     */
+    public Instant softDeleteFeedbackSession(String feedbackSessionName, String courseId)
+            throws EntityDoesNotExistException {
+        assert courseId != null;
+        assert feedbackSessionName != null;
+
+        FeedbackSession feedbackSessionEntity = getFeedbackSession(feedbackSessionName, courseId);
+
+        if (feedbackSessionEntity == null) {
+            throw new EntityDoesNotExistException(ERROR_UPDATE_NON_EXISTENT);
+        }
+
+        feedbackSessionEntity.setDeletedAt(Instant.now());
+        merge(feedbackSessionEntity);
```

Comment:
do we need to call `merge` here? I think `feedbackSessionEntity` is a managed object, so changes should be automatically made to the DB after a session flush

Reply:
Hi @daongochieu2810, for adding the `merge`, it does not immediately invoke the SQL statement directly, but registers the FeedbackSession into Hibernate's persistent state and the update will be flushed. 
There shouldnt be a difference between including `merge` and not since the FeedbackSession from the get method should already be in persistent state. Though adding the `merge` is potentially more defensive since if the get method is modified to return a transient state object, it will still set this transient state object as persistent as intended.